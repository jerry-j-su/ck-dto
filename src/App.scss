@use 'sass:color';
@use './theme.scss';

.App {
    min-width: 680px;
    height: 100vh;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    align-items: stretch;
    padding: 10rem 20rem;
    background-color: theme.$dark-bg;
}

.App-header {
    font-weight: 700;
}

.App-main {
    border: 1px solid color.scale(theme.$bg-front, $whiteness: 10%);
    border-radius: .5rem;
    padding: 2rem 5rem;
    min-height: 0;
    display: flex;
    flex-direction: column;
    align-items: stretch;
    background-color: theme.$bg-front;
}
