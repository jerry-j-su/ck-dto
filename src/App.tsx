import React, { useEffect } from 'react';
import useOrderFlow from './hooks/useOrderFlow'
import SearchBox from './components/SearchBox';
import OrderList from './components/OrderList';
import './App.scss';

function App() {
    const { connectOrderFlowSocket } = useOrderFlow()
    useEffect(() => connectOrderFlowSocket(), [connectOrderFlowSocket])

    return (
        <div className="App">
            <header className="App-header"></header>
            <main className="App-main">
                <SearchBox />
                <OrderList />
            </main>
        </div>
    );
}

export default App;
