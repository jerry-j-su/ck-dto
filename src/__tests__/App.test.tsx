import React from 'react'
import { render } from '@testing-library/react'
import { readFileSync  } from 'fs';

import App from '../App'

// module mocks
const sideEffectList: Function[] = []
jest.mock('react', () => {
    const originalModule = jest.requireActual('react')

    //Mock the default export and named export 'foo'
    return {
        __esModule: true,
        ...originalModule,
        //   default: jest.fn(() => 'mocked baz'),
        useRef: () => ({ current: { value: 'random content' } }),
        useState: (initialValue: any) => [initialValue, () => {}],
        useCallback: () => () => {},
        useEffect: (cb: Function) => sideEffectList.push(cb),
    }
})
const mockConnectOrderFlowSocket = jest.fn()
jest.mock('../hooks/useOrderFlow', () => ({
    __esModule: true,
    default: () => ({
        connectOrderFlowSocket: mockConnectOrderFlowSocket,
    })
}))
// data mocks
const mockOrders = JSON.parse(readFileSync(`${__dirname}/dummyOrders.json`, { encoding: 'utf-8' }))

describe('App.tsx', () => {
    test('App basic render', () => {
        const nodes = render(App())
        expect(nodes.container).toMatchSnapshot()
    })

    test('Connect order flow socket upon App starts', () => {
        // @ts-ignore
        expect(mockConnectOrderFlowSocket.mock.calls.length).toBe(0)
        if (sideEffectList[0]) sideEffectList[0]()
        // @ts-ignore
        expect(mockConnectOrderFlowSocket.mock.calls.length).toBe(1)

    })
})
