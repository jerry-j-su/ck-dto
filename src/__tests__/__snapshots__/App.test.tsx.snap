// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx App basic render 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    />
    <main
      class="App-main"
    >
      <section
        class="search-container"
      >
        <div
          class="search-input"
          tabindex="0"
        >
          <input
            type="combobox"
            value=""
          />
        </div>
      </section>
      <section
        class="order-list-container"
      >
        <dl>
          <dl>
            Count: 
          </dl>
          <dl>
            Filter Count: 
          </dl>
        </dl>
        <ul
          class="order-list-body"
        />
      </section>
    </main>
  </div>
</div>
`;
